### PR TITLE
[bitnami/*] Drop support for deprecated cert-manager annotation

### PR DESCRIPTION
**Pull request by pablogalegoc**
**Description of the change**

We are using the `ingress.certManager` parameter to include the `kubernetes.io/tls-acme: "true"` annotation in the Ingress resource to trigger the autogeneration of certificates cert-manager. This annotation comes from [kube-lego](https://github.com/jetstack/kube-lego), which is deprecated. We should instead point the user to add the [currently supported annotations](https://cert-manager.io/docs/usage/ingress/#supported-annotations).

**Benefits**

cert-manager annotations will be just another `ingress.annotations`

**Possible drawbacks**

We are removing the parameter from the `values.yaml` but not from the templates so it is usable in cases where it is still declared. However, the reference in the templates should be removed in the next major.

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/airflow/templates/_helpers.tpl [hunk 1]
```diff
@@ -460,3 +460,14 @@ Ref: https://github.com/bitnami/charts/pull/6096#issuecomment-856499047
...
+{{/*
+Returns whether the deprecated ingress.*.certManager boolean parameter is used or not
+This helper is needed in order to guarantee backwards-compatibility
+Ref: https://cert-manager.io/docs/usage/ingress/#supported-annotations
+*/}}
+{{- define "airflow.ingress.certManager.isLegacy" -}}
+{{- if typeIs "bool" .Values.ingress.certManager }}
+  true
+{{- else -}}{{- end -}}
+{{- end -}}
```
Comment (carrodher) [inline bitnami/airflow/templates/_helpers.tpl:473]:
```suggestion
{{- if typeIs "bool" .Values.ingress.certManager }}
    {{- true -}}
{{- end -}}
{{- end -}}
```

**Review by carrodher — COMMENTED** [review]
Minor suggestion. I think this is a good solution in order to make the change backward compatible. As discussed, adding it to the common can be the proper solution to reuse the code, I'm sure we can find a "category" to add this template or just create a miscellany one

**Comment by pablogalegoc** [comment]
Per @juan131 suggestion I'm going to explore removing the parameter and relying on the default `ingress.annotations` for this. Pros: We don't add new logic to handle backwards compatibility, we treat the `cert-manager` annotations as any other ingress annotation.

**Comment by pablogalegoc** [comment]
@carrodher done! If you think this is also a valid approach I'll just use this same PR to update the other instances of `ingress.certManager`

**Comment by carrodher** [comment]
For me it's fine! Just a minor comment, I am not sure if the README was generated by using the `readme-generator-tool` since it seems the table is not properly aligned (in terms of markdown, in the UI it looks fine).

Using the automated tool it should be aligned and you can see if the metadata in the _values.yaml_ was properly set

**Comment by pablogalegoc** [comment]
Yep! It was modified with `readme-generator`. I think it's the UI in Github's review section that is messing something, in [raw format](https://raw.githubusercontent.com/bitnami/charts/003c086225de52bb46c2b53eb05cb6c9cb585757/bitnami/airflow/README.md) you can see it is aligned

**Comment by pablogalegoc** [comment]
@carrodher this is a huge PR but basically is just:
- Deprecate `ingress.certManager` by commenting it
- Add a reference to the cert-manager annotations in the `ingress.annotations` documentation
- Add a mention to cert-manager in the `ingress.annotations` metadata so it appears in the README

**Review by juan131 — CHANGES_REQUESTED** [review]
I'm okay with the changes... That said, these 3 guides from Chart docs should be also updated @pablogalegoc 

- https://github.com/bitnami/charts-docs/blob/main/charts/cert-manager/_secure_ingress_resources.md.erb
- https://github.com/bitnami/charts-docs/blob/main/common/_enable_tls_ingress.md.erb
- https://github.com/bitnami/charts-docs/blob/main/charts/kubeapps/_expose_service.md.erb

I'll block this PR until there's another PR updating the chart docs.

---

Also, I remember from a private conversation that you were planning to create a "named template" to manage an existing condition in the ingress rule related to this `ingress.certManager`  parameter.. This was supposed to be done in the `_helpers.tpl` for Ghost, Kubeapps, Odoo, RabbitMQ and Redmine. Is this happening in other  PR?

**Comment by pablogalegoc** [comment]
Thanks @juan131! Yes, this is just the first batch of the easier ones. On the JIRA issue I'm keeping track of the progress. I'll open another PR similar to this for the remaining ones and another for Ghost, Kubeapps, Odoo, RabbitMQ and Redmine. This PR is already huge, I didn't want to make it worse 😅

**Comment by pablogalegoc** [comment]
The `readme-generator` CI check is failing for unrelated reasons, addressed at https://github.com/bitnami/charts/pull/7649

**Review by juan131 — APPROVED** [review]
LGTM

**Review by carrodher — APPROVED** [review]
LGTM

**Comment by AntonOfTheWoods** [comment]
It looks like I am not getting a certificate request after the update. Should the following be generating a `certificaterequest`? 

```
helm get values harbor -n harbor --all 
...
ingress:
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt-prod
    ingress.kubernetes.io/proxy-body-size: "0"
    ingress.kubernetes.io/ssl-redirect: "true"
    nginx.ingress.kubernetes.io/proxy-body-size: "0"
    nginx.ingress.kubernetes.io/ssl-redirect: "true"
  apiVersion: ""
  controller: default
  enabled: true
  hosts:
    core: registry.my.doma.in
    notary: notary.harbor.domain
  pathType: ImplementationSpecific
...
```

If not, can someone point me to some docs that might help out (the `readme` is not explicit enough for me unfortunately)? `cert-manager.io/cluster-issuer: letsencrypt-prod` seems to work in other charts' ingress so I'm a little unclear as to the issue here. Thanks.

**Pull request merged**